index.js

```diff
@@ -58,13 +58,45 @@ module.exports = (src, dest, opts) => {
 	return promise;
 };
 
-module.exports.sync = function (src, dest, opts) {
-	if (!src || !dest) {
-		throw new CpFileError('`src` and `dest` required');
+const checkSrcIsFile = function (src) {
+	const stat = fs.statSync(src);
+	if (stat.isDirectory()) {
+		throw Object.assign(new CpFileError(`EISDIR: illegal operation on a directory '${src}'`), {
+			errno: -21,
+			code: 'EISDIR',
+			src
+		});
 	}
+};
 
-	opts = Object.assign({overwrite: true}, opts);
+const fixupTimestampAndClose = function (read, write, src, dest) {
+	const stat = fs.fstatSync(read, src);
+	fs.futimesSync(write, stat.atime, stat.mtime, dest);
+	fs.closeSync(read);
+	fs.closeSync(write);
+};
+
+const copyNative = function (src, dest, opts) {
+	checkSrcIsFile(src);
+	fs.makeDirSync(path.dirname(dest));
```

Comment:
Do the read stuff before this so we don't unnecessarily create directories.

Reply:
yea, this is what `checkSrcIsFile` is for

Reply:
Just keep this line here and remove `checkSrcIsFile` since `fs.copyFile` handles directories.